src/main/java/com/flipkart/zjsonpatch/InPlaceApplyProcessor.java

```diff
@@ -147,9 +147,9 @@ public void replace(List<String> path, JsonNode value) {
             String fieldToReplace = path.get(path.size() - 1).replaceAll("\"", "");
             if (isNullOrEmpty(fieldToReplace) && path.size() == 1)
                 target = value;
-            else if (parentNode.isObject())
-                ((ObjectNode) parentNode).put(fieldToReplace, value);
```

Comment:
This method is deprecated now, but internally new 'replace' method does the same: puts a node to the parent. 
I wonder if issue should be also reported in Jackson library?